qiskit/circuit/parametertable.py

```diff
@@ -79,3 +84,209 @@ def __len__(self):
 
     def __repr__(self):
         return 'ParameterTable({})'.format(repr(self._table))
+
+
+def _deprecated_set_method():
+    def deprecate(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            # warn only once
+            if not wrapper._warned:
+                warnings.warn(f'The ParameterView.{func.__name__} method is deprecated as of '
+                              'Qiskit Terra 0.17.0 and will be removed no sooner than 3 months '
+                              'after the release date. Circuit parameters are returned as View '
+                              'object, not set. To use set methods you can explicitly cast to a '
+                              'set.', DeprecationWarning, stacklevel=2)
+                wrapper._warned = True
+            return func(*args, **kwargs)
+        wrapper._warned = False
+        return wrapper
+    return deprecate
+
+
+class ParameterView(MappingView):
+    """Temporary class to transition from a set return-type to list.
+
+    Derives from a list but implements all set methods, but all set-methods emit deprecation
+    warnings.
+    """
+
+    def __init__(self, iterable=None):
+        if iterable is not None:
+            self.data = list(iterable)
+        else:
+            self.data = []
+
+        super().__init__(self.data)
+
+    @_deprecated_set_method()
+    def add(self, x):
+        """Add a new element."""
+        if x not in self.data:
+            self.data.append(x)
+
+    @_deprecated_set_method()
+    def copy(self):
+        """Copy the ParameterView."""
+        return self.__class__(self.data.copy())
```

Comment:
`List`s also implement `copy`, so should this be deprecated?

Reply:
Good point!